internal/inventory/gcpfetcher/fetcher_assets.go

```diff
@@ -172,3 +194,173 @@ func appendIfExists(slice []string, fields map[string]*structpb.Value, key strin
+	if err := mapstructure.Decode(labels.GetStructValue().AsMap(), &labelsMap); err != nil {
+		return nil
+	}
+
+	return labelsMap
+}
+
+var assetEnrichers = map[string]func(item *gcpinventory.ExtendedGcpAsset, fields map[string]*structpb.Value) []inventory.AssetEnricher{
+	gcpinventory.IamRoleAssetType:               noopEnricher,
+	gcpinventory.CrmFolderAssetType:             noopEnricher,
+	gcpinventory.CrmProjectAssetType:            noopEnricher,
+	gcpinventory.StorageBucketAssetType:         noopEnricher,
+	gcpinventory.IamServiceAccountKeyAssetType:  noopEnricher,
+	gcpinventory.CloudRunService:                noopEnricher,
+	gcpinventory.CrmOrgAssetType:                enrichOrganization,
+	gcpinventory.ComputeInstanceAssetType:       enrichComputeInstance,
+	gcpinventory.ComputeFirewallAssetType:       enrichFirewall,
+	gcpinventory.ComputeSubnetworkAssetType:     enrichSubnetwork,
+	gcpinventory.IamServiceAccountAssetType:     enrichServiceAccount,
+	gcpinventory.GkeClusterAssetType:            enrichGkeCluster,
+	gcpinventory.ComputeForwardingRuleAssetType: enrichForwardingRule,
+	gcpinventory.CloudFunctionAssetType:         enrichCloudFunction,
+}
+
+func enrichOrganization(_ *gcpinventory.ExtendedGcpAsset, fields map[string]*structpb.Value) []inventory.AssetEnricher {
+	return []inventory.AssetEnricher{
+		inventory.WithOrganization(inventory.Organization{
+			Name: getStringValue("displayName", fields),
+		}),
+	}
+}
+
+func enrichComputeInstance(item *gcpinventory.ExtendedGcpAsset, fields map[string]*structpb.Value) []inventory.AssetEnricher {
+	return []inventory.AssetEnricher{
+		inventory.WithCloud(inventory.Cloud{
+			// This will override the default Cloud fields, so we re-add the common ones
+			Provider:         inventory.GcpCloudProvider,
+			AccountID:        item.CloudAccount.AccountId,
+			AccountName:      item.CloudAccount.AccountName,
+			ProjectID:        item.CloudAccount.OrganisationId,
+			ProjectName:      item.CloudAccount.OrganizationName,
+			ServiceName:      item.AssetType,
+			InstanceID:       getStringValue("id", fields),
+			InstanceName:     getStringValue("name", fields),
+			MachineType:      getStringValue("machineType", fields),
+			AvailabilityZone: getStringValue("zone", fields),
+		}),
+		inventory.WithHost(inventory.Host{
+			ID: getStringValue("id", fields),
+		}),
+	}
+}
+
+func enrichFirewall(_ *gcpinventory.ExtendedGcpAsset, fields map[string]*structpb.Value) []inventory.AssetEnricher {
+	return []inventory.AssetEnricher{
+		inventory.WithNetwork(inventory.Network{
+			Name:      getStringValue("name", fields),
+			Direction: getStringValue("direction", fields),
+		}),
+	}
+}
+
+func enrichSubnetwork(_ *gcpinventory.ExtendedGcpAsset, fields map[string]*structpb.Value) []inventory.AssetEnricher {
+	return []inventory.AssetEnricher{
+		inventory.WithNetwork(inventory.Network{
+			Name: getStringValue("name", fields),
+			Type: strings.ToLower(getStringValue("stackType", fields)),
```

Comment:
Do we need to lower the stack type here? Why not keep it consistent with what we get from upstream?